### DISCUSSION TITLE
Relicensing to MIT/Apache-2.0

### DISCUSSION THREAD

**Discussion by arxanas** [arxanas/git-branchless#813] [General]
Hi everyone, as you may know, git-branchless was licensed as GPL-2.0-only so that it could use Sapling SCM's DAG implementation (see https://github.com/arxanas/git-branchless/pull/65). It was always my intention to license [the program source code itself more permissively](https://github.com/arxanas/git-branchless/pull/65#issuecomment-890621805), but the resulting program (when linked against a GPL library) becomes GPL-licensed as a whole. This is confusing to explain and implement, so I simply licensed the whole program under GPL 2.0.

The DAG library is [licensed as MIT as of v0.3.0](https://docs.rs/crate/esl01-dag/0.3.0/source/LICENSE). After upgrading, we will no longer have any GPL dependencies. I am proposing that we relicense git-branchless to a dual MIT/Apache-2.0 license, so that we can reuse components in other systems such as [Jujutsu](https://github.com/martinvonz/jj) (i.e. copy and paste some code one day). I've already started writing smaller modules licensed under MIT/Apache-2.0 to that end.

To relicense the program source code, I need the permission of those who made substantial code contributions. This approach comes from some other GNU project that did a big relicensing effort, whose name I can't remember.

I defined a "substantial code contribution" as 10 commits or a net positive 100 lines of code, only including changes to the program source code, not associated build code or scripts. Please don't take it personally if you are excluded — I appreciate even the small contributions! — but I picked the criterion arbitrarily and only then went to look at the corresponding users 🙂. This criterion includes these users:

- [x] @arxanas: [commits](https://github.com/arxanas/git-branchless/commits?author=arxanas) / [permission granted](https://github.com/arxanas/git-branchless/discussions/813)
- [x] @claytonrcarter: [commits](https://github.com/arxanas/git-branchless/commits?author=claytonrcarter) / [permission granted](https://discord.com/channels/968932220549103686/969291218347524238/1050942196431470703)
- [x] @waych [commits](https://github.com/arxanas/git-branchless/commits?author=waych) / [permission granted](https://github.com/arxanas/git-branchless/pull/65#issuecomment-890608314)
- [x] @bcongdon [commits](https://github.com/arxanas/git-branchless/commits?author=bcongdon) / [permission granted](https://github.com/arxanas/git-branchless/discussions/813#discussioncomment-5276383)
- [x] @wabain [commits](https://github.com/arxanas/git-branchless/commits?author=wabain) / [permission granted](https://github.com/arxanas/git-branchless/discussions/813#discussioncomment-5058599)
- [x] @bcspragu [commits](https://github.com/arxanas/git-branchless/commits?author=bcspragu) / [permission granted](https://github.com/arxanas/git-branchless/discussions/813#discussioncomment-5058177)
- [x] @rslabbert [commits](https://github.com/arxanas/git-branchless/commits?author=rslabbert) / [permission granted](https://github.com/arxanas/git-branchless/discussions/813#discussioncomment-5058283)

[Full contributor stats](https://github.com/arxanas/git-branchless/graphs/contributors)

To move forward:

- If you are on the list above and comfortable relicensing to MIT/Apache-2.0, please comment your assent below.
- If you are a contributor and would like to decline the relicensing of your contribution or have other comments, please let me know.
- In accordance with the way that the one other GNU project did it, if I don't hear back from contributors after some period of time, I will eventually assume their assent by default.

Thanks everyone for your contributions to git-branchless!

**Comment by bcspragu** [comment]
@bcspragu commenting in assent of relicensing to MIT/Apache-2.0

**Comment by rslabbert** [comment]
@rslabbert assent of relicensing to MIT/Apache-2.0

**Comment by wabain** [comment]
@wabain assenting to relicensing to MIT/Apache-2.0

**Comment by bcongdon** [comment]
@bcongdon assenting to relicensing to MIT/Apache-2.0
